runtime/v2/module.go

```diff
@@ -246,3 +244,7 @@ func ProvideGenesisTxHandler(appBuilder *AppBuilder) genesis.TxHandler {
 func ProvideCometService() comet.Service {
 	return &services.ContextAwareCometInfoService{}
 }
+
+func ProvideAppVersionModifier(app *AppBuilder) app.VersionModifier {
+	return app.app
```

Comment:
I would return nil here (and add a comment saying only baseapp implements a version modifier -- https://github.com/cosmos/cosmos-sdk/blob/0a6ee406a02477ae8ccbfcbe1b51fc3930087f4c/x/upgrade/keeper/keeper.go#L438), and remove the other methods on the app